src/i2c.rs

```diff
@@ -495,19 +498,19 @@ macro_rules! hal {
                             self.nb.i2c.ctl0.modify(|_, w| w.acken().set_bit());
                         }
                         2 => {
-                            self.nb.i2c.ctl0.modify(|_, w| w.pecen().set_bit().acken().set_bit());
+                            self.nb.i2c.ctl0.modify(|_, w| w.acken().clear_bit());
                             self.nb.i2c.stat0.read();
                             self.nb.i2c.stat1.read();
-                            self.nb.i2c.ctl0.modify(|_, w| w.acken().clear_bit());
 
                             busy_wait_cycles!(wait_for_flag!(self.nb.i2c, btc), self.data_timeout)?;
-                            self.nb.send_stop();
+                            busy_wait_cycles!(wait_for_flag!(self.nb.i2c, rbne), self.data_timeout)?;
                             buffer[0] = self.nb.i2c.data.read().trb().bits();
+                            busy_wait_cycles!(wait_for_flag!(self.nb.i2c, rbne), self.data_timeout)?;
                             buffer[1] = self.nb.i2c.data.read().trb().bits();
+                            self.nb.send_stop();
 
                             busy_wait_cycles!(self.nb.wait_for_stop(), self.data_timeout)?;
-                            self.nb.i2c.ctl0.modify(|_, w| w.pecen().clear_bit().acken().clear_bit());
-                            self.nb.i2c.ctl0.modify(|_, w| w.acken().set_bit());
+                            self.nb.i2c.ctl0.modify(|_, w| w.poap().clear_bit().acken().set_bit());
```

Comment:
Is this going to cause an ACK to be sent? I'm not an expert on i2c, it just looks like a difference compared to what was there before.